### PR TITLE
Add "static_lambda" CS fixer rule

### PR DESCRIPTION
**Pull request by phansys**
## Add "static_lambda" CS fixer rule

I am targeting this branch, because these changes are fully BC.

### CODE REVIEW

**Comment by OskarStark** [comment]
We need to update the phpcs dist file in the dev-kit repo to have the config in all bundles

File: src/Resources/skeleton/Admin.tpl.php [hunk 1]
```diff
@@ -1,28 +1,28 @@
-<?= "<?php\n" ?>
+<?= "<?php\n"; ?>
```
Comment (greg0ire) [inline src/Resources/skeleton/Admin.tpl.php]: How does this relate to your changes?

Comment (phansys) [reply src/Resources/skeleton/Admin.tpl.php]: These changes don't are related to the new rule, of course. They are the result of executing `php-cs-fixer` with the existing configuration. I could try to execute the fixer just filtering for the new rule, but I think the codebase should be kept up to date with the provided configuration.
I await for your concerns.

Comment (greg0ire) [reply src/Resources/skeleton/Admin.tpl.php]: Oh you mean running the fixer right now without changes will give that result?

**Comment by phansys** [comment]
Here for `dev-kit`: sonata-project/dev-kit#491.

File: src/Resources/skeleton/Admin.tpl.php (hunk 1 above)
Comment (phansys) [reply src/Resources/skeleton/Admin.tpl.php]: That's right.

Comment (greg0ire) [reply src/Resources/skeleton/Admin.tpl.php]: … I tried locally, and those fixes don't appear for me… I upgraded php-cs-fixer first (I'm using 2.14.2)

**Comment by greg0ire** [comment]
You can remove the changelog unless this fixes a bug / introduces an improvement that can be perceived by the end user.

File: src/Resources/skeleton/Admin.tpl.php (hunk 1 above)
Comment (greg0ire) [reply src/Resources/skeleton/Admin.tpl.php]: Anyway I'm ok with the changes

Comment (phansys) [reply src/Resources/skeleton/Admin.tpl.php]: I have the same version:
```
$ composer global show friendsofphp/php-cs-fixer -N
Changed current directory to /home/jspagnoletti/.composer
name     : friendsofphp/php-cs-fixer
descrip. : A tool to automatically fix PHP code style
keywords : 
versions : * v2.14.2
type     : application
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
source   : [git] https://github.com/FriendsOfPHP/PHP-CS-Fixer.git ff401e58261ffc5934a58f795b3f95b355e276cb
dist     : [zip] https://api.github.com/repos/FriendsOfPHP/PHP-CS-Fixer/zipball/ff401e58261ffc5934a58f795b3f95b355e276cb ff401e58261ffc5934a58f795b3f95b355e276cb
path     : /home/jspagnoletti/.composer/vendor/friendsofphp/php-cs-fixer
names    : friendsofphp/php-cs-fixer
```
This is my local result in the latest commit (af8723a2b657e4bff1a902edefa251304e3b558c) from `3.x` branch:
```
$ php-cs-fixer fix . --config=.php_cs.dist -v
Loaded config default from ".php_cs.dist".
Using cache file ".php_cs.cache".
Paths from configuration file have been overridden by paths provided as command arguments.
................................................................F........................F..F..................................................................................................................................................................................................................................................FF.
Legend: ?-unknown, I-invalid file syntax, file ignored, S-Skipped, .-no changes, F-fixed, E-error
   1) tests/Fixtures/Admin/ModelAdmin.php (header_comment)
   2) tests/Fixtures/Controller/BatchAdminController.php (strict_comparison)
   3) tests/Fixtures/Controller/ModelAdminController.php (header_comment)
   4) src/Resources/skeleton/AdminController.tpl.php (semicolon_after_instruction)
   5) src/Resources/skeleton/Admin.tpl.php (semicolon_after_instruction)

Fixed all files in 29.760 seconds, 28.000 MB memory used
```

**Comment by phansys** [comment]
Changelog removed.

File: src/Resources/skeleton/Admin.tpl.php (hunk 1 above)
Comment (greg0ire) [reply src/Resources/skeleton/Admin.tpl.php]: Ah I see, I'm using `make cs-fix`, which in turn does `php-cs-fixer fix --verbose`. Using `.` in your command overrides the config about paths.

Comment (phansys) [reply src/Resources/skeleton/Admin.tpl.php]: Oh, my bad! Thank you for pointing me the issue. Changes are updated.

Comment (greg0ire) [reply src/Resources/skeleton/Admin.tpl.php]: No problem :)

File: .php_cs.dist [hunk 2]
```diff
@@ -41,11 +42,11 @@ $rules = [
...
     '@PHP71Migration' => true,
     '@PHP71Migration:risky' => true,
-    'compact_nullable_typehint' => true,
     'void_return' => null,
+    'php_unit_strict' => true,
+    'static_lambda' => true,
     'strict_comparison' => true,
     'strict_param' => true,
-    'php_unit_strict' => true,
 ];
```
Comment (OskarStark) [inline .php_cs.dist]: Please keep the order like it is and only add the new option/rule

Comment (phansys) [reply .php_cs.dist]: Done.

**Comment by OskarStark** [comment]
I think this can be closed due to https://github.com/sonata-project/SonataAdminBundle/pull/5533, right?

**Comment by phansys** [comment]
Yes! Closing. Thank you.

**Comment by OskarStark** [comment]
We have to thank you for this nice contribution 👌🏻🙃

**Pull request closed without merging**